### ISSUE TITLE
Please add config support to error on unawaited promise.catch() or promise.then()

### ISSUE THREAD

**Issue by Woodz** [SebastienGllmt/eslint-plugin-no-floating-promise#18]
Hi maintainers 👋

First off, thank you for your work on this project — `eslint-plugin-no-floating-promise` has been really helpful for maintaining async safety in our codebase!

---

### 🧠 Feature Request

I'd like to propose either:

- a **new config option** for the current rule (`no-floating-promise`), or  
- a **separate rule** (e.g. `no-unawaited-promise` or `no-unsequenced-promise`)

…that **flags unawaited Promises even if they're chained with `.then()` or `.catch()`**.

---

### ✅ Current Behavior

I understand that the current rule intentionally treats `.then()` and `.catch()` as valid "handling" of the promise — and that makes sense for many use cases, especially when promises are fire-and-forget or explicitly chained.

```ts
doSomethingAsync().catch(console.error); // ✅ allowed by default
```

---

### 🚫 Proposed Rule Behavior

In some environments (e.g. critical workflows or testing), it’s important to ensure that **all Promises are explicitly awaited**, regardless of whether `.catch()` is attached — to guarantee ordering and error propagation.

The following would ideally be flagged:

```ts
// ❌ should be flagged (new rule behavior)
doSomethingAsync().catch(console.error);

// ✅ explicitly awaited
await doSomethingAsync().catch(console.error);

// ✅ explicitly fire-and-forget
void doSomethingAsync().catch(console.error);
```

---

### 🔧 Why a Separate Rule or Option?

I understand this goes beyond what `no-floating-promise` was designed for, so I think it might be more appropriate as either:

- an **optional strict mode** for `no-floating-promise`, or  
- a **separate rule entirely**, like `no-unawaited-promise`, to avoid breaking existing setups.

---

### 🙏 Summary

This change would help enforce a stricter style where `await` is always required unless explicitly ignored, which improves clarity and reduces bugs related to async execution ordering — even when a `.catch()` is attached.

Let me know your thoughts — happy to discuss further or help explore a PR if it’s viable.

Thanks again!
